### PR TITLE
feat (group): add apis to manage users

### LINKED ISSUE #62: feat: api to assign users to a group
Current:
- We dont have any API to assign users to a team as team_member, only users who create the team using API are assigned as team_admin

Expectations:
- we should be able to assign a user to a group/team as team_member

### PR DESCRIPTION
**Pull request by rsbh**
Close #62 and #55

### CODE REVIEW

File: internal/authz/spicedb/spicedb.go [hunk 1]
```diff
@@ -72,7 +72,7 @@ func (p Permission) AddRelation(ctx context.Context, relation model.Relation) er
 	request := &pb.WriteRelationshipsRequest{
 		Updates: []*pb.RelationshipUpdate{
 			{
-				Operation:    pb.RelationshipUpdate_OPERATION_CREATE,
+				Operation:    pb.RelationshipUpdate_OPERATION_TOUCH,
```
Comment (krtkvrm) [inline internal/authz/spicedb/spicedb.go:75]: why touch?

File: store/postgres/groups.go [hunk 2]
```diff
@@ -22,11 +24,20 @@ type Group struct {
...
-	getGroupsQuery    = `SELECT id, name, slug, org_id, metadata, created_at, updated_at from groups where id=$1;`
-	listGroupsQuery   = `SELECT id, name, slug, org_id, metadata, created_at, updated_at from groups`
-	updateGroupQuery  = `UPDATE groups set name = $2, slug = $3, org_id = $4, metadata = $5, updated_at = now() where id = $1 RETURNING id, name, slug, org_id, metadata, created_at, updated_at;`
+var (
+	createGroupsQuery   = `INSERT INTO groups(name, slug, org_id, metadata) values($1, $2, $3, $4) RETURNING id, name, slug, org_id, metadata, created_at, updated_at;`
+	getGroupsQuery      = `SELECT id, name, slug, org_id, metadata, created_at, updated_at from groups where id=$1;`
+	listGroupsQuery     = `SELECT id, name, slug, org_id, metadata, created_at, updated_at from groups`
+	updateGroupQuery    = `UPDATE groups set name = $2, slug = $3, org_id = $4, metadata = $5, updated_at = now() where id = $1 RETURNING id, name, slug, org_id, metadata, created_at, updated_at;`
+	listGroupUsersQuery = fmt.Sprintf(
+		`SELECT u.id as id, u."name" as name, u.email as email, u.metadata as metadata, u.created_at as created_at, u.updated_at as updated_at
```
Comment (krtkvrm) [inline store/postgres/groups.go:33]: should we try using query builder here ?

File: store/postgres/migrations/20220207071931_add-relations-namespace-id.up.sql [hunk 3]
```diff
@@ -0,0 +1,5 @@
+ALTER TABLE relations
+    ADD COLUMN namespace_id VARCHAR REFERENCES namespaces (id),
+    DROP CONSTRAINT IF EXISTS relations_subject_namespace_id_subject_id_role_id_object_na_key;
+
+CREATE UNIQUE INDEX unique_relation_with_ns_id ON relations (subject_namespace_id, subject_id, object_namespace_id, object_id, COALESCE(role_id, ''), COALESCE(namespace_id, ''))
```
Comment (krtkvrm) [inline store/postgres/migrations/20220207071931_add-relations-namespace-id.up.sql:5]: end with `;`

File: store/postgres/relation.go [hunk 4]
```diff
@@ -78,16 +88,51 @@ const (
...
+		       subject_id, 
+		       object_namespace_id, 
+		       object_id, 
+		       role_id,
+		       namespace_id,
+		       created_at, 
+		       updated_at 
+		FROM relations 
+		WHERE subject_namespace_id=$1 AND subject_id=$2 AND object_namespace_id=$3 AND object_id=$4 AND (role_id IS NULL OR role_id = $5) AND (namespace_id IS NULL OR namespace_id = $6)`
+	deleteRelationById = `DELETE FROM relations WHERE id = $1`
```
Comment (krtkvrm) [inline store/postgres/relation.go:108]: end with `;`

File: internal/authz/spicedb/spicedb.go (hunk 1 above)
Comment (rsbh) [reply internal/authz/spicedb/spicedb.go:75]: TOUCH is upsert, with CREATE I was getting error for duplicate entries.
https://buf.build/authzed/api/file/main/authzed/api/v1/core.proto#L65

File: store/postgres/groups.go (hunk 2 above)
Comment (rsbh) [reply store/postgres/groups.go:33]: sure, we can look into query builder while cleanup and refactoring.

**Review by krtkvrm — APPROVED** [review]

**Pull request merged**